README.md

```diff
@@ -1,10 +1,326 @@
+Now that one has successfully trained and evaluated a DeepRacer model using the notebook, one can explore how to customize training in a variety of areas in this section.
+
+This section will show one what areas to modify for customizations; once one makes the modifications one will need to go back and re-run the appropriate sections of the notebook to apply the modifications. It is intended as a general guidebook for one to pursue their own path for customization, not a prescriptive set of steps. Feel free to "think big" and brainstorm on the possibilities.
+
+Create a reward function
+
+In general, you design your reward function to act like an incentive plan. You can customize your reward function with relevant input parameters passed into the reward function.
+Reward function files
+
+1. Explore to src/artifacts/rewards/ in the notebook to see example reward functions.
+
+1. Once you pick or modify one, locate the cell in the notebook labeled Copy custom files to S3 bucket so that sagemaker & robomaker can pick it up and modify this line as appropriate to copy your new reward function appropriately.
+2. !aws s3 cp ./src/artifacts/rewards/default.py {s3_location}/customer_reward_function.py
+
+Example reward functions
+Follow the Center Line in Time Trials
+follow_center_line.py
+
+
+This example determines how far away the agent is from the center line, and gives higher reward if it is closer to the center of the track, encouraging the agent to closely follow the center line.
+
+Stay Inside the Two Borders in Time Trials
+stay_inside_two_border.py
+This example simply gives high rewards if the agent stays inside the borders, and let the agent figure out what is the best path to finish a lap. It is easy to program and understand, but likely takes longer to converge.
+
+Prevent Zig-Zag in Time Trials
+prevent_zig_zag.py
+This example incentivizes the agent to follow the center line but penalizes with lower reward if it steers too much, which helps prevent zig-zag behavior. The agent learns to drive smoothly in the simulator and likely keeps the same behavior when deployed in the physical vehicle.
+
+Stay On One Lane without Crashing into Stationary Obstacles or Moving Vehicles
+object_avoidance_head_to_head.py
+This reward function rewards the agent to stay between the track borders and penalizes the agent for getting too close to the next object in the front. The agent can move from lane to lane to avoid crashes. The total reward is a weighted sum of the reward and penalty. The example gives more weight to the penalty term to focus more on safety by avoiding crashes. You can play with different averaging weights to train the agent with different driving behaviors and to achieve different driving performances.
+
+Create your own reward function
+
+If you wish to create your own reward function there is a pattern to the function that you must use.
+
+
+12345def reward_function(params) :
+
+    reward = ...
+
+    return float(reward)
```

Comment:
Check the syntax of this code element, during review it wasn't showing correctly as a code sample

Reply:
Done.